### PR TITLE
Short-code generator for PublicLab.org 

### PR DESCRIPTION
**Pull request by sagarpreet-chadha**
Fixes https://github.com/publiclab/leaflet-environmental-layers/issues/167

* [ ] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

### CODE REVIEW

**Comment by sagarpreet-chadha** [comment]
**Inline Code** :` [map:layers:23:77:mapKnitter,odorReport,wisconsin,asian,clouds]`

Screenshot 2019-08-24 at 4 24 30 PM



* All these layers are ON by default as shown in pic .

**Comment by sagarpreet-chadha** [comment]
Failing for unrelated test : 

Screenshot 2019-08-24 at 4 37 43 PM

@IshaGupta18 @cesswairimu ?

**Comment by sagarpreet-chadha** [comment]
Tests passed 🎉

**Comment by sagarpreet-chadha** [comment]
Once merged, i will make new note or maybe update existing https://github.com/publiclab/plots2/pull/6204 one . 

So we can have layers in this format only : `[map:layers:23:77:mapKnitter,odorReport,wisconsin,asian,clouds]`

### Layers name :
 "purpleAirMarker",
"skyTruth",
"fracTracker",
"odorReport",
"mapKnitter",
"toxicRelease"
 "Power",
"Petroleum",
"Telecom",
"Water"
"wisconsin"
,"fracTrackerMobile"
income"
,"americanIndian"
,"asian",
"black",
"multi","
hispanic",
"nonWhite",
"white",
"plurality"
      "clouds",
"cloudsClassic",
"precipitation",
"precipitationClassic",
"rain"
"rainClassic","
snow",
"pressure",
"pressureContour"
,"temperature"
,"wind"
,"city"

File: app/models/concerns/node_shared.rb [hunk 1]
```diff
@@ -202,6 +202,21 @@ def self.people_map(body, _page = 1)
     end
   end
 
+  # [map:layers:_latitude_:_longitude:SkyTruth,MapKnitter]
+  def self.layers_map(body, _page = 1)
+    body.gsub(/\[map\:layers\:(\S+)\:(\S+)\:(\w+)((\,\w+)*)]/) do |_tagname|
+      lat = Regexp.last_match(1)
+      lon = Regexp.last_match(2)
+      primaryLayer =  Regexp.last_match(3).to_s
+      secondoryLayers = Regexp.last_match(4).to_s
```
Comment (seabl) [inline app/models/concerns/node_shared.rb]: Typo in the above line?

**Comment by sagarpreet-chadha** [comment]
Okay, if the first layer is any of the current inline map name, and after
that all layers can be LEL layers . I guess we can integrate then . Let me
try this out  .

**Comment by jywarren** [comment]
awesome!

**Comment by sagarpreet-chadha** [comment]
Only LEL : 
`[map:layers::23:77:mapKnitter,odorReport,wisconsin,asian,clouds]`

People + LEL :
`[map:layers:people:23:77:mapKnitter,odorReport,wisconsin]`

Content + LEL
`[map:layers:content:23:77:mapKnitter,odorReport,wisconsin]`

Tag + LEL
`[map:layers:tag:infragram:23:77:skyTruth,mapKnitter]`

**Comment by sagarpreet-chadha** [comment]
@jywarren ...how does this looks?

**Comment by jywarren** [comment]
Wow this is super impressive!

**Comment by jywarren** [comment]
Let me review in more depth... sorry, just a little bit longer!

**Comment by sagarpreet-chadha** [comment]
# LEL 2.0 Released and integrated here! 

* So now to use LEL we just have to : 
```
 L.LayerGroup.EnvironmentalLayers({
        include: ['mapknitter'],
        hash: true,
      }).addTo(map);
```

* We need to merge yarn.lock file which is in this PR .

* Inline map Syntax : 

 Only LEL : 
`[map:layers::23:77:mapKnitter,odorReport,wisconsin,asian,clouds]`

 People + LEL :
`[map:layers:people:23:77:mapKnitter,odorReport,wisconsin]`

 Content + LEL
`[map:layers:content:23:77:mapKnitter,odorReport,wisconsin]`

 Tag + LEL
`[map:layers:tag:infragram:23:77:skyTruth,mapKnitter]`

**Comment by sagarpreet-chadha** [comment]
###

**Comment by sagarpreet-chadha** [comment]
@jywarren ...

`Sprockets::FileNotFound: couldn't find file 'debounce/index.js' with type 'application/javascript'`

It passed for commit #75dbf5e0d8be66753856a657303549b9e9cf148a , after that i just rebased with master .

**Comment by cesswairimu** [comment]
That's  strange.I have restarted travis @sagarpreet-chadha maybe it will disappear  :crossed_fingers: .

**Comment by cesswairimu** [comment]
:thinking: @sagarpreet-chadha does `yarn install` run successfully locally for u? After the rebase?

**Comment by cesswairimu** [comment]
I am guessing it might be a problem with yarn.lock or package.json maybe a hierarchy thing .

**Comment by sagarpreet-chadha** [comment]
Hi yes yarn gives no error for me . Strange indeed!

**Comment by sagarpreet-chadha** [comment]
Okay failing for unrelated system test.
Screenshot 2019-09-08 at 12 52 57 AM

**Comment by sagarpreet-chadha** [comment]
Tests are passing yayyy 🎉

**Comment by cesswairimu** [comment]
:tada: :tada:

**Comment by IshaGupta18** [comment]
Finally! I have been following all the mails :P

**Comment by jywarren** [comment]
Oh awesome!!!! Great, reading through now!

File: app/assets/javascripts/leaflet_helper.js [hunk 2]
```diff
@@ -47,172 +34,119 @@
...
-    L.control.layers(baseMaps,overlayMaps).addTo(map);
+   function setupInlineLEL(map , layers, mainLayer) {
+       layers = layers.split(',');
+
+       L.tileLayer('https://a.tiles.mapbox.com/v3/jywarren.map-lmrwb2em/{z}/{x}/{y}.png', {
+           attribution: '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
+       }).addTo(map) ;
+
+       L.LayerGroup.EnvironmentalLayers({
+           include: layers,
```
Comment (jywarren) [inline app/assets/javascripts/leaflet_helper.js]: ❤️❤️❤️❤️ so nice!!!!

**Comment by jywarren** [comment]
This is fantastic. Great work. Can you test it out at https://stable.publiclab.org once it builds, to double-check? THANK YOU and congrats!

**Comment by sagarpreet-chadha** [comment]
@jywarren ,
Seems to work great: https://stable.publiclab.org/notes/chadha_sagarpreet97/09-14-2019/new-inline-map-testing !

**Comment by jywarren** [comment]
This is live! However, something odd is going on in wiki pages. I think
perhaps because of the rich text previewing or something, it is not
clickable. Is it possible it's being loaded twice?

**Comment by sagarpreet-chadha** [comment]
Okay wierd, works fine in LocalHost.
Trying to figure it out.

**Comment by sagarpreet-chadha** [comment]
The popup works if there is only 1 inline map.

**Comment by sagarpreet-chadha** [comment]
Hey @jywarren , i figured out the problem and the solution ❤️ 

I have made PR in `leaflet` repository: https://github.com/Leaflet/Leaflet/pull/6838
Let's wait for their response :)

**Pull request merged**